.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,69 @@
+name: Linting & tests
+
+on: [push, pull_request]
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    name: Lint - Type consistency (mypy)
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+      - name: Install with linting tools
+        run: pip install .[linting]
```

Comment:
It might be too much, but we could also consider adding separate setups for mypy, isort, and flake8, so that they can be installed with `pip install .[mypy]` and so on, instead of the `[.linting]` each time.

Reply:
The balance in existing CI and here too is intended to be between keeping dependencies centralized (`setup.py` only; some systems list their deps as explicit installs in CI) and not too fine-grained for clutter in that file. I agree we could in principle add more extras. I think one original motivation for having linting separate was just that `mypy` doesn't work so well under some testing environments, so `.[testing]` would fail - and incidentally this was fewer dependencies to install in each CI run.